sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -591,6 +605,8 @@ class RapidsBufferCatalog(
         if (!bufferHasSpilled) {
           // if the spillStore specifies a maximum size spill taking this ceiling
           // into account before trying to create a buffer there
+          // TODO: we may need to handle what happens if we can't spill anymore
```

Comment:
What happens now if we cannot get enough room. The long term plan is to have a fallback where we go directly to the disk and bypass the memory store.

Reply:
Right now it's just going to log it and not spill. If I have all the host buffers as non-spillable, it's going to say "oh well, I didn't spill... lets just add more". The next time around, if those host buffers are spillable then I have something to move.

Part of this is as bad as today but part of it is worse. Today host buffers don't know about spillability, so we could "spill" things to disk and they were being held by application code so we didn't really free anything. The part that is worse than today is that before we would, as soon as the application code is done with the host buffer, free that buffer that was spilled, since the application held that last reference.

We could disable spillability for now for host buffers that way is the least bad ;) 

Or I could work on the "skip the host store, go to disk change". Though that feels like a different PR.

Reply:
This is fine for now. Just wanted to understand it. We are not likely to run into this in the short term.